### PR TITLE
Add 'visible' to layer options

### PR DESCRIPTION
**Pull request by rubenmoya**
Related to: https://github.com/CartoDB/carto.js/issues/2004

### Description
This PR adds `visible` to the layer options so people can create layers that are hidden by default.

It also improves the examples a little bit.

### CODE REVIEW

File: src/api/v4/layer/layer.js [hunk 1]
```diff
@@ -63,26 +81,33 @@ var metadataParser = require('./metadata/parser');
...
-function Layer (source, style, options) {
+function Layer (source, style, options = {}) {
   Base.apply(this, arguments);
 
-  options = options || {};
-
   _checkSource(source);
   _checkStyle(style);
 
+  options = _.defaults(options, {
```
Comment (jesusbotella) [inline src/api/v4/layer/layer.js]: Why did you change the default assignment to `_.defaults`? 🤔

Comment (jesusbotella) [reply src/api/v4/layer/layer.js]: I know that the `visible` check _might not be_ as easy as the other ones, but you can take the chance to make a simple type check, so that you only allow boolean values.

Something like that:
```js
this._visible = _.isBoolean(options.visible) ? options.visible : true;
```

Comment (rubenmoya) [reply src/api/v4/layer/layer.js]: It was easier to handle the default value of `visible` being it a boolean, since I can't do `falsy || true` and I didn't want to add a bunch of conditionals, what do you recommend?

Comment (jesusbotella) [reply src/api/v4/layer/layer.js]: I recommend what I posted on top of your comment 😂

Comment (rubenmoya) [reply src/api/v4/layer/layer.js]: you were faster pressing enter 😂

**Comment by jesusbotella** [comment]
Working like a charm 🚀
